### PR TITLE
Support for HEAD requests

### PR DESCRIPTION
**Pull request by masterspambot**
#47

### CODE REVIEW

File: katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourceHead.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import io.katharsis.request.path.JsonPath;
+import io.katharsis.request.path.PathIds;
+import io.katharsis.request.path.ResourcePath;
+import io.katharsis.resource.exception.ResourceNotFoundException;
+import io.katharsis.resource.include.IncludeLookupSetter;
+import io.katharsis.resource.registry.RegistryEntry;
+import io.katharsis.resource.registry.ResourceRegistry;
+import io.katharsis.resource.registry.responseRepository.ResourceRepositoryAdapter;
+import io.katharsis.response.BaseResponseContext;
+import io.katharsis.response.JsonApiResponse;
```
Comment (sonarqube-reporter) [inline katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourceHead.java]: MINOR Remove this unused import 'io.katharsis.response.JsonApiResponse'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourceHead.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import io.katharsis.response.ResourceResponseContext;
+import io.katharsis.utils.parser.TypeParser;
+
+import java.io.Serializable;
```
Comment (sonarqube-reporter) [inline katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourceHead.java]: MINOR Remove this unused import 'java.io.Serializable'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourceHead.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package io.katharsis.dispatcher.controller.resource;
+
+import io.katharsis.dispatcher.controller.HttpMethod;
+import io.katharsis.queryParams.QueryParams;
+import io.katharsis.repository.RepositoryMethodParameterProvider;
+import io.katharsis.request.dto.RequestBody;
+import io.katharsis.request.path.JsonPath;
+import io.katharsis.request.path.PathIds;
```
Comment (sonarqube-reporter) [inline katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourceHead.java]: MINOR Remove this unused import 'io.katharsis.request.path.PathIds'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourceHead.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import io.katharsis.request.path.ResourcePath;
+import io.katharsis.resource.exception.ResourceNotFoundException;
```
Comment (sonarqube-reporter) [inline katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourceHead.java]: MINOR Remove this unused import 'io.katharsis.resource.exception.ResourceNotFoundException'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourceHead.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import io.katharsis.resource.include.IncludeLookupSetter;
+import io.katharsis.resource.registry.RegistryEntry;
```
Comment (sonarqube-reporter) [inline katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourceHead.java]: MINOR Remove this unused import 'io.katharsis.resource.registry.RegistryEntry'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourceHead.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import io.katharsis.resource.registry.ResourceRegistry;
+import io.katharsis.resource.registry.responseRepository.ResourceRepositoryAdapter;
```
Comment (sonarqube-reporter) [inline katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourceHead.java]: MINOR Remove this unused import 'io.katharsis.resource.registry.responseRepository.ResourceRepositoryAdapter'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by sonarqube-reporter** [comment]
SonarQube analysis reported 7 issues
* CRITICAL 1 critical
* MINOR 6 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [RelationshipsResourceUpsert.java#L83](https://github.com/katharsis-project/katharsis-framework/blob/1ae055f2b79e203765afd9294769d353e5a2d740/katharsis-core/src/main/java/io/katharsis/dispatcher/controller/relationship/RelationshipsResourceUpsert.java#L83): The Cyclomatic Complexity of this method "handle" is 12 which is greater than 10 authorized. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by chb0github** [comment]
This is a full implementation?

**Comment by chb0github** [comment]
I can only assume it works. 

But, now it has merge conflicts :-/

**Comment by jfleck1** [comment]
Hey guys,I am fairly new to Katharsis. We're looking for HEAD support. Is this PR ready? Need any help?

**Comment by chb0github** [comment]
The PR is old and has merge conflicts. If you wanna tackle it you have my
support

**Pull request closed without merging**